components.md

```diff
@@ -3,91 +3,61 @@
-Actions which the actor can take could be anything from measuring on a regular basis ("acting" as a sensor), commanding hardware devices through the use of the driver to do X or Y, processing some data and re-emitting results (e.g. a PID controller).
-
-Potentially a GUI could be attached here too.
+A Actor is a component that interfaces with a (hardware) Device and that has a specific API on the ECP side.
+A Actor must contain a Driver object dealing with communication with the Device.
 
-The Actor would send commands to a Driver, instructing it to `set`/`get`/`call` a given Parameter/Action by name, with a value if appropriate, and receive replies.
+We define how the other ECP components interact with the Actor, how it determines and announces its capabilities, etc.
+An Actor implements the mapping between ECP messages and Driver calls/attribute access.
 
-This could look for example like this:
-:::{mermaid}
-sequenceDiagram
-    par
-        Actor1->>Driver42: GET temp_K
-        Driver42-->>HW-on-COM1: "*TEMP?\n"
-        HW-on-COM1-->>Driver42: "275.14\n"
-        Driver42->>Actor1: 275.14
-    and
-        Actor1->>Driver11: GET temperature
-        Note over Driver11: Fetched a fresh value 23ms ago, using cached value
-        Driver11->>Actor1: 300.1
-    end
-    Note over Actor1: computes average temperature, converts to degC
-    Note over Actor1: stores/caches value maybe
-    Actor1->>Observer1: "Average oven temperature" 14.47 degC
-:::
-
-:::{admonition} TODO
-We could achieve more complete separation and make the Actor purely about processing _inputs_, doing some _operation_ and generating _outputs_.
-This could be either stateless (temperature conversion) or stateful (PID control).
-Maybe a better name would be "Processor" then.
-This means that _all_ device-related stuff stays compartmentalized in the Driver, and an Actor would address that Driver over the network (to fetch the Parameter values it is interested in).
-
-Thoughts?
-:::
-
-## Driver
-A Driver is a component that interfaces with a (hardware) Device in a manner we don't specify, and that has a specific API on the ECP side.
-A driver must contain some object dealing with hardware communication.
-This may be a `pymeasure.Instrument` instance or something from another instrument library.
-This is the place where all instrument libraries (including pymeasure) wire their hardware interface classes into ECP.
-
-Concerning the ECP, we draw the abstraction boundary at the Driver -- the details on how this communicates with a Device (SCPI, dll, ...) should not be relevant for the protocol details.
-What we define is how the other ECP components interact with the Driver, how it determines and announces its capabilities, etc.
-
-A driver must contain/manage
-* An object that communicates with the connected Device, including managing its lifecycle (init, operations, shutdown)
+A Actor must contain/manage/provide
```

Comment:
```suggestion
An Actor must contain/manage/provide
```

Next line should start with "A Driver"

Reply:
Ha, that's what you get for doing search/replace and not switching off the mental autocorrect 😅